sdk/core/Azure.Core/tests/TestFramework/FailureSimulationTransport.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using NUnit.Framework;
+
+namespace Azure.Core.Testing
+{
+    public class FailureSimulationTransport : HttpPipelineTransport
+    {
+        private readonly HttpPipelineTransport _innerTransport;
+        private readonly SimulateFailureAttribute _simulateFailure;
+        private readonly Dictionary<string, int> _failedMessages;
+
+        public FailureSimulationTransport(HttpPipelineTransport innerTransport)
+        {
+            _innerTransport = innerTransport;
+            _simulateFailure = SimulateFailureAttribute.Current;
```

Comment:
I don't think the layering here might cause issues later. It's impossible for this type to be configured or used outside of NUnit test and also impossible to have different settings inside one test.